src/linker/implementation.py

```diff
@@ -3,27 +3,24 @@
 
 from loguru import logger
 
+from linker.configuration import Config
 from linker.step import Step
 from linker.utilities.data_utils import load_yaml
 from linker.utilities.slurm_utils import get_slurm_drmaa
-from linker.utilities.spark_utils import build_cluster
+from linker.utilities.spark_utils import build_spark_cluster
 
 
 class Implementation:
     def __init__(
         self,
+        config: Config,
         step: Step,
-        implementation_name: str,
-        implementation_config: Optional[Dict[str, Any]],
-        container_engine: str,
-        resources: Dict[str, Any],
     ):
         self.step = step
+        self.config = config
         self._pipeline_step_name = step.name
-        self.name = implementation_name
-        self.config = self._format_config(implementation_config)
-        self._container_engine = container_engine
-        self.resources = resources
+        self.name = config.get_implementation_name(step.name)
+        self.implementation_config = self._get_implementation_config()
```

Comment:
Having both `implmentation.config` and `implementation.implementation_config` is confusing. Do we need both? Is there a way to make this clearer?

Reply:
The idea is that "config" is the well-known Config object. "implementation_config" is NOT that, but rather implementation-specific configurations that get brought in through the pipeline.yaml (e.g. `DUMMY_CONTAINER_BROKEN: true` or `DUMMY_CONTAINER_INCREMENT: 5`).

We previously didn/t have this problem b/c self.config was only the implementation configuration. But now that I'm passing in the Config object (to push the resources call to when we actually use it), it caused the confusion.

Reply:
As discussed, I'll do this refactor as part of a separate PR to unblock @patricktnast from merging his PR